### PR TITLE
Adding a new search app with specific filter for the Perovskite Database entries

### PR DESCRIPTION
**Pull request by Pepe-Marquez**
## Summary by Sourcery

Add a new search application for the Perovskite Solar Cell Database with specific filtering capabilities, update the schema for better data handling, and ensure the application is correctly integrated and tested.

New Features:
- Introduce a new search application for the Perovskite Solar Cell Database, allowing users to search and filter database entries with specific criteria.

Enhancements:
- Change the data type of the 'band_gap_graded' field from string to boolean in the Perovskite schema to improve data accuracy and consistency.

Build:
- Add a new entry point for the Perovskite Solar Cell Database application in the project's configuration file.

Tests:
- Add a test to ensure the new Perovskite Solar Cell Database application can be imported without errors, verifying the integrity of the pydantic model validation.

### CODE REVIEW

**Comment by Pepe-Marquez** [comment]
@lauri-codes: this PR adds a new search page for the perovskite database with custom filters. I might add an additional section later, but I believe it's ready for you to start reviewing from your perspective. It should work with the current solar cell entries.

The PR also includes a search page for the ions database that we’ve compiled. To test the ions entries, you can drop the .xlsx file into the test/data folder. Additionally, I’ve updated the category of the apps.

Regarding the possibility of having the plots in the left panel: for instance, in the Publication section, we could have a plot of efficiency vs. publication date, with other topical plots in subsequent sections. This way, users would have pre-configured plots under different categories that they could push to the main dashboard with the plus button, serving as both a functional feature and a source of inspiration. However, I’m completely fine if you think this approach isn’t ideal.

**Review by lauri-codes — COMMENTED** [review]
Few small comments, also played around with the apps a bit.

In "The Perovskite Solar Cell Database" you are loading 4 scatter plots into the dashboard: this takes a while, maybe test a bit to see that the initial loading time is reasonable.

In general, I see that you are using the menus more as an extension of the dashboard: maybe we should see if that is in general a direction where we want to go.

File: src/perovskite_solar_cell_database/apps/perovskite_solar_cell_database_app.py [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+                    ),
+                ],
+            ),
+            Menu(
+                title='Perovskite Material',
+                size=MenuSizeEnum.XXL,
+                items=[
+                    MenuItemPeriodicTable(
+                        quantity='results.material.elements',
+                        show_input=True,
```
[the remark this answers is not in the document]
Comment (Pepe-Marquez) [reply src/perovskite_solar_cell_database/apps/perovskite_solar_cell_database_app.py:126]: Removed the `show_input`

File: src/perovskite_solar_cell_database/apps/perovskite_solar_cell_database_app.py [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+        Column(quantity='published', label='Access'),
+    ],
+    menu=Menu(
+        items=[
+            Menu(
+                title='Publication',
+                size=MenuSizeEnum.XXL,
+                items=[
+                    MenuItemTerms(
+                        search_quantity='data.ref.journal#perovskite_solar_cell_database.schema.PerovskiteSolarCell',
```
[the remark this answers is not in the document]
Comment (Pepe-Marquez) [reply src/perovskite_solar_cell_database/apps/perovskite_solar_cell_database_app.py]: Done

**Comment by Pepe-Marquez** [comment]
@lauri-codes, I think it would be good to see how the 4 plots behave in production. I think if you agree we can try to deploy this and see. If not, I can try to reduce it to one or two plots in te dashboard. Would it be ok if we merge and try to deploy? @hampusnasstrom, is it also ok with you to dump the ions database in staging?

**Comment by hampusnasstrom** [comment]
I made a few small fixes in the llm-extraction-branch so I will pull those here as well.

**Comment by lauri-codes** [comment]
Yes, you can just deploy this and see how the scatter plots behave themselves in the oasis distro.

**Review by hampusnasstrom — APPROVED** [review]

**Pull request merged**